File: 02_12_2024_Check_If_a_Word_Occurs_As_a_Prefix_of_Any_Word_in_a_Sentence.java
import java.util.StringTokenizer;

class Solution {
    public int isPrefixOfWord(String sentence, String searchWord) {
        StringTokenizer tokenizer = new StringTokenizer(sentence);
        int n = searchWord.length();
        int t = 0;
        
        while (tokenizer.hasMoreTokens()) {
            String word = tokenizer.nextToken();
            t++;
            if (searchWord.equals(word.substring(0, Math.min(n, word.length())))) {
                return t;
            }
        }
        return -1;
    }
}

File: 02_12_2024_Check_If_a_Word_Occurs_As_a_Prefix_of_Any_Word_in_a_Sentence.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        stringstream s(sentence);
        string word;
        int n = searchWord.size();
        int t = 0;
        while(s>>word)
        {
            t++;
            if(searchWord == word.substr(0,n))
            {
                return t;
            }
        }
        return -1;
    }
};
